pkg/kp/service.go

```diff
@@ -7,6 +7,12 @@ import (
 	"github.com/square/p2/pkg/pods"
 )
 
+var (
+	// prints the HTTP response on stderr, while exiting 0 if the status code is 200
+	httpStatusCheck  = `if [[ $(curl http://$(hostname):%v/_status -s -o /dev/stderr -w "%%{http_code}") == "200" ]] ; then exit 0 ; else exit 2; fi`
+	httpsStatusCheck = `if [[ $(curl https://$(hostname):%v/_status -s -o /dev/stderr -w "%%{http_code}" --cacert ${SECRETS_PATH}/service2service.ca.pem) == "200" ]] ; then exit 0 ; else exit 2; fi`
```

Comment:
That's exactly what I was worried about, but at this point I don't think it's worth the code to find a better, more general solution (I am totally open to ideas though). Especially because consul has to have access to whatever certificate you specify. If the pod manifest gets to name the cert to use, then consul has to have read rights on it... gets out of hand pretty quickly for arbitrary certificates.